### ISSUE TITLE
Running error of invalid size of input

### ISSUE THREAD

**Issue by karenyun** [KD-TAO/DyCoke#4]
Hi, thanks for your excellent work.

When running the code, I tried only to use one GPU, then I got the following error:

<img width="1496" alt="Image" src="https://github.com/user-attachments/assets/3ccf3637-6734-40e4-becb-4b2da4f61319" />

I checked the `inputs_embeds` passed to the `super().generate()` function within the `generate()` method in llava_qwen.py. The shape of inputs_embeds is [1, 3152, 896].

However, the error traceback points to the `super().forward()` function, where inputs_embeds is None and the shape of input_ids is (1, 0).

I'm curious about this behavior: why does the `super().generate()` call eventually lead to the forward() function, even though the superclass Qwen2Model does not define a `generate()` method?

I’d appreciate any insights you can share about this error. Thank you!

**Comment by KD-TAO** [comment]
Thank you for your question. This problem has not occurred in my operation. Can you provide more information about your eval script?  A definition of the `generate()` function can be found under the` LlavaQwenForCausalLM`

**Comment by karenyun** [comment]
Hi @KD-TAO Thanks so much for your quick reply!

I used this eval.sh:

```
CUDA_VISIBLE_DEVICES=0 accelerate launch \
    --num_processes=1 \
    --main_process_port=25001 \
    -m lmms_eval \
    --model llava_onevision \
   --model_args pretrained=lmms-lab/llava-onevision-qwen2-0.5b-ov,conv_template=qwen_1_5,model_name=llava_qwen,device_map=auto,dycoke=True \
    --tasks activitynetqa \
    --batch_size 1 \
    --log_samples \
    --log_samples_suffix llava_onevision \
    --output_path ./logs/
```

Yes, I found the `generate()` function is defined in  `LlavaQwenForCausalLM`, but the end of this function is also called the super.generate(). I did not find the super() class `Qwen2ForCausalLM` and `LlavaMetaForCausalLM` define the generate() function. I am wondering about it.

**Comment by KD-TAO** [comment]
I think the actual definition of the `generate` function should be in the library, but the problem you mentioned should be irrelevant. I have already reproduced the problem you mentioned. I suggest you check whether the version of `transformers` library is **4.40.0**. I found that the problem you encountered would occur after upgrading to **4.48.0**.

**Comment by karenyun** [comment]
Thanks so much!
Yes, after upgrading `transformers` to version 4.40.0, it worked.
I found that the reason transformers was upgraded beyond 4.40.0 was due to this command `pip install lmms-eval`.
Could you please share the correct version of `lmms-eval` to install, so that it doesn’t automatically upgrade other packages?

**Comment by karenyun** [comment]
Hi,

After switching to using two GPUs and setting `num_processes=2`, the program gets stuck at the following line of code:
`gathered_item = lm.accelerator.gather(instances_rnk).cpu().detach().numpy().tolist()`
This corresponds to line `428` in `lmms_eval/evaluator.py.`

I'm wondering if this issue might be related to incompatible versions of `accelerate `or `deepspeed`, causing the code to not run out of this line.
Could you please advise on what might be causing this and how to resolve it?

Thanks!

**Comment by karenyun** [comment]
Hi, I have solved the issues. Thanks so much!
